README.md

```diff
@@ -1 +1,84 @@
-# autogluon_bench
\ No newline at end of file
+<div align="left">
+  <img src="https://user-images.githubusercontent.com/16392542/77208906-224aa500-6aba-11ea-96bd-e81806074030.png" width="350">
+</div>
+
+# AutoGluon-Bench
+
+## Setup
+
+```
+git clone https://github.com/suzhoum/autogluon-bench.git
+cd autogluon-bench
+git checkout poc_0.0.1
+
+# create virtual env
+python3.9 -m venv .venv
+source .venv/bin/activate
+
+# install autogluon-bench
+pip install -e .
+```
+
+## Run benchmarkings locally
+
+Currently, `tabular` makes use of the [AMLB](https://github.com/openml/automlbenchmark) benchmarking framework, so we will supply the required and optional arguments when running benchmark for tabular. It can only run on the stable and latest(master) build of the framework at the moment. In order to benchmark on a custom branch, changes need to be made on AMLB.
+
+```
+python ./runbenchmarks.py  --config_file path/to/local_config_file
+```
+A sample local config file is available for reference at `./sample_configs/local_configs.yaml`. All keys should have single value.
+
+Note that `multimodal` benchmarking directly calls the `MultiModalPredictor` without the extra layer of [AMLB](https://github.com/openml/automlbenchmark), so the set of arguments we call is different from that of running `tabular`. Currently, we support benchmarking `multimodal` on custom branch of the main repository or any forked repository.
+
+To customize the benchmarking experiment, including adding more hyperparameters, and evaluate on more metrics, you can refer to `./src/autogluon/bench/frameworks/multimodal/exec.py`.
+
+Results are saved under `$WORKING_DIR/benchmark_runs/$module/{$benchmark_name}_{$timestamp}`
+
+
+## Run benchmarkings on AWS
+
+The infrastructure is built on AWS CDK, where a AWS Batch compute environment is setup to run the benchmarkings. 
+
+In order to run AWS CDK and build containers, the following setup is required to install [Node.js](https://nodejs.org/) and [AWS CDK](https://docs.aws.amazon.com/cdk/v2/guide/getting_started.html#getting_started_install)
+
+```
+curl https://raw.githubusercontent.com/creationix/nvm/master/install.sh | bash  # replace bash with other shell (e.g. zsh) if you are using a different one
+source ~/.bashrc
+nvm install 18.13.0  # install Node.js
+npm install -g aws-cdk  # install aws-cdk
+cdk --version  # verify the installation
+```
+
+The default configurations of the infrastructure is located at `./src/autogluon/bench/cloud/aws/default_config.yaml`.
+```
+CDK_DEPLOY_ACCOUNT: dummy
+CDK_DEPLOY_REGION: dummy
+PREFIX: ag-bench-test
+RESERVED_MEMORY_SIZE: 15000
+MAX_MACHINE_NUM: 20
+BLOCK_DEVICE_VOLUME: 100
+INSTANCE: g4dn.2xlarge
+METRICS_BUCKET: autogluon-benchmark-metrics
+DATA_BUCKET: automl-mm-bench
+VPC_NAME: automm-batch-stack/automm-vpc
+LAMBDA_FUNCTION_NAME: ag-bench-test-job-function
```

Comment:
Added an [issue](https://github.com/autogluon/autogluon-bench/issues/4). The design of providing custom `VPC_NAME` is because we wanted to reuse vpc ideally. I can do it in a follow up PR to make it optional, and also auto generate `LAMBDA_FUNCTION_NAME`. Created an [issue](https://github.com/autogluon/autogluon-bench/issues/6).

Reply:
+1. Make most of the config optional and provide default values. Power users should still be able to customize themselves